enrichment-mongo/src/main/java/eu/europeana/enrichment/mongo/dao/TopicEntityDaoImpl.java

```diff
@@ -0,0 +1,99 @@
+package eu.europeana.enrichment.mongo.dao;
+
+import static dev.morphia.query.experimental.filters.Filters.eq;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import dev.morphia.Datastore;
+import dev.morphia.query.internal.MorphiaCursor;
+import eu.europeana.enrichment.common.commons.AppConfigConstants;
+import eu.europeana.enrichment.model.TopicEntity;
+import eu.europeana.enrichment.model.TopicModel;
+import eu.europeana.enrichment.model.impl.TopicEntityImpl;
+import eu.europeana.enrichment.model.impl.TopicModelImpl;
+
+@Repository(AppConfigConstants.BEAN_ENRICHMENT_TOPIC_ENTITY_DAO)
+public class TopicEntityDaoImpl implements TopicEntityDao{
+
+	
+
+	@Autowired
+	private Datastore enrichmentDatastore;
+	
+	@Override
+	public void saveTopicEntity(TopicEntity topicEntity) {
```

Comment:
The implementation of this method would need to be simplified, e.g. like in this example: https://github.com/EnrichEuropeana/enrichment/blob/55d8d8bdb5a5c03d5e3d573801689678ea7ae9b7/enrichment-mongo/src/main/java/eu/europeana/enrichment/mongo/dao/TranslationEntityDaoImpl.java#L102, and before calling the method, we would need to make sure the topicEntity provided as a parameter is the one from the db (if it exists in the db). In that way, the existing topic entity in the db will be updated, and not the new one created.

Reply:
this is done at the upper level, check how the method is being used

Reply:
rename to save